sdks/python/apache_beam/transforms/core.py

```diff
@@ -2364,6 +2364,22 @@ def from_runner_api_parameter(
   def runner_api_requires_keyed_input(self):
     return True
 
+  def get_windowing(self, inputs):
```

Comment:
Do I understand right that the idea is for it all to be lazily propagated?

Reply:
It's been a while since I've touched this code, but I think so. Can't be hard to verify.